In [1]:
#importing necessary library
import pandas as pd


In [2]:
#reading the file in
weather = pd.read_csv("merged_dataframe.csv")

In [3]:
#printing the dataframe
weather.head()

state  year  yearly_avg_snow  yearly_max_temp  yearly_min_temp  \
0  Alaska  2005         6.230833             73.5            -22.7   
1  Alaska  2006         7.095000             70.1            -29.7   
2  Alaska  2007         6.613333             75.1            -25.8   
3  Alaska  2008         8.601667             70.2            -23.0   
4  Alaska  2009         8.861667             78.6            -24.4   

   yearly_avg_temp  yearly_avg_prcp  yearly_min_prcp  yearly_max_prcp  \
0        35.053333         3.210000             1.59             6.21   
1        32.066667         2.837500             0.99             5.69   
2        33.525833         2.635000             0.76             5.82   
3        31.105833         2.935833             2.04             5.32   
4        32.727500         2.755833             1.28             4.56   

   avg_snow_summer  ...  min_temp_winter  avg_temp_summer  avg_temp_winter  \
0         1.460000  ...            -22.7        46.864286           18.518   
1         2.138571  ...            -29.7        45.245714           13.616   
2         1.244286  ...            -25.8        46.288571           15.658   
3         3.332857  ...            -23.0        43.210000           14.160   
4         1.438571  ...            -24.4        45.860000           14.342   

   avg_prcp_summer  avg_prcp_winter  total pop  in-state pop  percentage  \
0         3.174286            3.260     115525         84150   27.158624   
1         3.522857            1.878     139758        102086   26.955165   
2         2.912857            2.246     135095         98512   27.079463   
3         3.408571            2.274     150412        110205   26.731245   
4         2.674286            2.870     147525        107051   27.435350   

   top1  top1 state  
0  5711  Washington  
1  4850  Washington  
2  3985  California  
3  4951  California  
4  3780  California  

[5 rows x 24 columns]

In [4]:
#dropping data before 2010
weather = weather[~weather['year'].isin([2005, 2006, 2007,2008,2009])]

In [5]:
weather.head()

state  year  yearly_avg_snow  yearly_max_temp  yearly_min_temp  \
5  Alaska  2010         5.407500             72.0            -29.1   
6  Alaska  2011         7.995000             71.1            -22.3   
7  Alaska  2012         7.418333             72.2            -45.1   
8  Alaska  2013         6.785833             78.3            -22.5   
9  Alaska  2014         3.747500             69.5            -19.7   

   yearly_avg_temp  yearly_avg_prcp  yearly_min_prcp  yearly_max_prcp  \
5        33.590833         2.602500             1.19             4.31   
6        32.845833         3.048333             0.77             5.48   
7        30.401667         2.878333             1.09             7.38   
8        34.404167         3.345833             1.54             6.77   
9        36.409167         3.018333             1.10             5.15   

   avg_snow_summer  ...  min_temp_winter  avg_temp_summer  avg_temp_winter  \
5         1.952857  ...            -29.1        45.842857           16.438   
6         2.205714  ...            -22.3        45.227143           15.512   
7         1.611429  ...            -45.1        44.217143           11.060   
8         1.821429  ...            -22.5        45.801429           18.448   
9         1.107143  ...            -19.7        46.370000           22.464   

   avg_prcp_summer  avg_prcp_winter  total pop  in-state pop  percentage  \
5         2.758571            2.384     697235        660909    5.210008   
6         3.174286            2.872     707221        672137    4.960826   
7         3.270000            2.330     716579        683164    4.663129   
8         3.595714            2.996     718112        683752    4.784769   
9         3.285714            2.644     721116        683274    5.247699   

   top1  top1 state  
5  4123       Texas  
6  4548  Washington  
7  3725  Washington  
8  2984  Washington  
9  4841  Washington  

[5 rows x 24 columns]

In [6]:
#resetting the index
weather.reset_index(drop=True, inplace=True)

In [7]:
states = weather['state'].unique()

In [8]:
#checking the stats for variables
weather.describe()

year  yearly_avg_snow  yearly_max_temp  yearly_min_temp  \
count   588.000000       572.000000       588.000000       588.000000   
mean   2015.666667         2.717869        91.684694        14.915986   
std       3.707506         2.311846         6.994659        14.276359   
min    2010.000000         0.000000        69.500000       -45.100000   
25%    2012.750000         0.600208        86.800000         6.575000   
50%    2015.500000         2.438333        91.300000        15.400000   
75%    2018.250000         3.993750        95.800000        22.625000   
max    2022.000000        10.060833       112.500000        63.900000   

       yearly_avg_temp  yearly_avg_prcp  yearly_min_prcp  yearly_max_prcp  \
count       588.000000       588.000000       588.000000       588.000000   
mean         54.972795         2.997231         0.964252         6.226531   
std           8.610831         1.314866         0.744705         2.619946   
min          30.401667         0.397500         0.000000         0.850000   
25%          49.360417         1.998333         0.250000         4.640000   
50%          53.507917         3.168750         0.900000         6.140000   
75%          61.316042         3.865417         1.502500         7.750000   
max          77.690000         5.833333         2.990000        16.330000   

       avg_snow_summer  avg_snow_winter  ...  min_temp_summer  \
count       570.000000       569.000000  ...       588.000000   
mean          0.369456         6.039128  ...        36.244388   
std           0.780599         5.006141  ...        10.809857   
min           0.000000         0.000000  ...        -9.300000   
25%           0.000000         1.460000  ...        29.300000   
50%           0.028571         5.580000  ...        35.900000   
75%           0.385714         8.940000  ...        43.100000   
max           9.424286        24.132000  ...        68.500000   

       min_temp_winter  avg_temp_summer  avg_temp_winter  avg_prcp_summer  \
count       588.000000       588.000000       588.000000       588.000000   
mean         14.917517        66.466713        38.881310         3.312461   
std          14.281485         6.867832        11.605394         1.504336   
min         -45.100000        44.217143        11.060000         0.204286   
25%           6.575000        61.776071        31.159000         2.111786   
50%          15.400000        65.434286        37.311000         3.580000   
75%          22.625000        71.645714        45.881500         4.333929   
max          63.900000        80.745714        74.442000         7.564286   

       avg_prcp_winter     total pop  in-state pop  percentage           top1  
count       588.000000  5.880000e+02  5.880000e+02  588.000000     588.000000  
mean          2.555908  6.419247e+06  6.269839e+06    2.884193   24563.982993  
std           1.495127  7.037604e+06  6.925442e+06    0.956776   17150.747126  
min           0.198000  5.549150e+05  5.268690e+05    1.121510    2984.000000  
25%           1.239500  1.856363e+06  1.793305e+06    2.166180   12670.750000  
50%           2.575000  4.557198e+06  4.453296e+06    2.788637   20738.000000  
75%           3.575000  7.409158e+06  7.155490e+06    3.453064   31882.000000  
max           6.988000  3.882628e+07  3.833736e+07    6.238005  107546.000000  

[8 rows x 22 columns]

In [9]:
#printing the column names for those to drop
weather.columns


Index(['state', 'year', 'yearly_avg_snow', 'yearly_max_temp',
       'yearly_min_temp', 'yearly_avg_temp', 'yearly_avg_prcp',
       'yearly_min_prcp', 'yearly_max_prcp', 'avg_snow_summer',
       'avg_snow_winter', 'max_temp_summer', 'max_temp_winter',
       'min_temp_summer', 'min_temp_winter', 'avg_temp_summer',
       'avg_temp_winter', 'avg_prcp_summer', 'avg_prcp_winter', 'total pop',
       'in-state pop', 'percentage', 'top1', 'top1 state'],
      dtype='object')

In [10]:
#dropping insignificant variables
droppingcols = ["top1", "top1 state","avg_snow_summer", "avg_snow_winter", "max_temp_summer", "max_temp_winter", "min_temp_summer", "min_temp_winter", "avg_temp_summer","avg_temp_winter","avg_prcp_summer","avg_prcp_winter"]
weather= weather.drop(columns=droppingcols) #had to drop top columns they were not significant, the other columns had to be dropped because of multicollinearity (NAs in the summary of regression)

In [11]:
weather.head()

state  year  yearly_avg_snow  yearly_max_temp  yearly_min_temp  \
0  Alaska  2010         5.407500             72.0            -29.1   
1  Alaska  2011         7.995000             71.1            -22.3   
2  Alaska  2012         7.418333             72.2            -45.1   
3  Alaska  2013         6.785833             78.3            -22.5   
4  Alaska  2014         3.747500             69.5            -19.7   

   yearly_avg_temp  yearly_avg_prcp  yearly_min_prcp  yearly_max_prcp  \
0        33.590833         2.602500             1.19             4.31   
1        32.845833         3.048333             0.77             5.48   
2        30.401667         2.878333             1.09             7.38   
3        34.404167         3.345833             1.54             6.77   
4        36.409167         3.018333             1.10             5.15   

   total pop  in-state pop  percentage  
0     697235        660909    5.210008  
1     707221        672137    4.960826  
2     716579        683164    4.663129  
3     718112        683752    4.784769  
4     721116        683274    5.247699

In [12]:
#checking for nas in data
weather.isna().sum()

state               0
year                0
yearly_avg_snow    16
yearly_max_temp     0
yearly_min_temp     0
yearly_avg_temp     0
yearly_avg_prcp     0
yearly_min_prcp     0
yearly_max_prcp     0
total pop           0
in-state pop        0
percentage          0
dtype: int64

In [13]:
#replacing nas with 0s
weather['yearly_avg_snow'] = weather['yearly_avg_snow'].fillna(0)

### Linear regression

In [14]:
#importing library for models
from sklearn.preprocessing import LabelEncoder

#creating a new numerical based on state
label_encoder = LabelEncoder()
weather['state_encoded'] = label_encoder.fit_transform(weather['state'])

In [15]:
weather.head()

state  year  yearly_avg_snow  yearly_max_temp  yearly_min_temp  \
0  Alaska  2010         5.407500             72.0            -29.1   
1  Alaska  2011         7.995000             71.1            -22.3   
2  Alaska  2012         7.418333             72.2            -45.1   
3  Alaska  2013         6.785833             78.3            -22.5   
4  Alaska  2014         3.747500             69.5            -19.7   

   yearly_avg_temp  yearly_avg_prcp  yearly_min_prcp  yearly_max_prcp  \
0        33.590833         2.602500             1.19             4.31   
1        32.845833         3.048333             0.77             5.48   
2        30.401667         2.878333             1.09             7.38   
3        34.404167         3.345833             1.54             6.77   
4        36.409167         3.018333             1.10             5.15   

   total pop  in-state pop  percentage  state_encoded  
0     697235        660909    5.210008              1  
1     707221        672137    4.960826              1  
2     716579        683164    4.663129              1  
3     718112        683752    4.784769              1  
4     721116        683274    5.247699              1

In [16]:
#printing the final dataset datatypes
final = weather.drop(['state'],axis=1)
final.dtypes

year                 int64
yearly_avg_snow    float64
yearly_max_temp    float64
yearly_min_temp    float64
yearly_avg_temp    float64
yearly_avg_prcp    float64
yearly_min_prcp    float64
yearly_max_prcp    float64
total pop            int64
in-state pop         int64
percentage         float64
state_encoded        int64
dtype: object

In [17]:
#X, features for the models
X = weather.drop(["percentage",'state'], axis = 1)
X.dtypes

year                 int64
yearly_avg_snow    float64
yearly_max_temp    float64
yearly_min_temp    float64
yearly_avg_temp    float64
yearly_avg_prcp    float64
yearly_min_prcp    float64
yearly_max_prcp    float64
total pop            int64
in-state pop         int64
state_encoded        int64
dtype: object

In [18]:
#y, response variable for the models
y = weather['percentage']
y.dtypes

dtype('float64')

In [19]:
from sklearn.model_selection import train_test_split

In [20]:
#splitting the data using random split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2,random_state=1234)

In [21]:
#split by years used to do trial and error
# train_data = weather[(weather['year'] > 2009) & (weather['year'] <= 2019)]
# test_data = weather[weather['year'] > 2019]
#didn't split the data this way as the mse is higher this way

In [22]:
# X_train = train_data.drop(['percentage', 'state'], axis=1)
# y_train = train_data['percentage']

In [23]:
# X_test = test_data.drop(['percentage', 'state'], axis=1)
# y_test = test_data['percentage']

In [24]:
X_test.head()


year  yearly_avg_snow  yearly_max_temp  yearly_min_temp  yearly_avg_temp  \
585  2019         7.970000             90.2              0.9        42.965833   
29   2015         0.708333             94.4             21.5        62.676667   
270  2016         0.614167             91.1             17.3        58.365000   
100  2014         0.216667             93.3             25.7        64.366667   
278  2012         0.000000             94.1             35.5        66.460000   

     yearly_avg_prcp  yearly_min_prcp  yearly_max_prcp  total pop  \
585         1.085000             0.31             3.95     571502   
29          5.410000             1.26            11.79    2929305   
270         3.440833             0.63             7.91    6000269   
100         3.849167             2.46             6.57    9918670   
278         4.881667             1.48             8.28    2942684   

     in-state pop  state_encoded  
585        541255             48  
29        2847711              3  
270       5837460             23  
100       9644575              8  
278       2869184             22

In [25]:
#importing library for LR model
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error

In [26]:
#creating LR instance
linearreg = LinearRegression()

In [27]:
#fitting the model on train data
#test-train-random split is better
linearreg.fit(X_train, y_train)

LinearRegression()

In [28]:
#predicting y for test data
y_predreg = linearreg.predict(X_test)

In [29]:
#printing the predicted ys
y_predreg

array([ 3.8207643 ,  2.13530454,  2.92199246,  2.79005287,  2.13871738,
        1.77023111, -0.2325194 ,  3.76954466,  2.08670121,  2.20868854,
        2.35235746,  3.55985747,  1.74065098,  3.95497029,  4.42839977,
        1.24376673,  3.54222457,  4.36979039,  2.99663595,  3.07091355,
        3.2517475 ,  4.07525789,  2.33214208,  2.83361782,  3.13192803,
        1.82889931,  4.11192456,  3.43883626,  1.57374596,  1.06190255,
        2.7098696 ,  3.05458168,  3.16838415,  3.39580774,  3.11152809,
        4.22946757,  3.42657614,  2.27779479,  3.4343234 ,  2.30337981,
        3.77250934,  2.50933627,  2.81583114,  2.86224439,  2.90683374,
        2.99444248,  3.73661428,  3.30271954,  1.74688473,  2.40911194,
        2.63868462,  2.86329993,  3.04663412,  2.93052334,  2.93755879,
        4.00548158,  2.70680566,  3.2669251 ,  2.52224708,  3.33310578,
        2.43506721,  3.41092626,  3.119174  ,  3.48810428,  2.89149661,
        3.40799453,  3.22940365,  2.5677237 ,  2.29182777,  2.76

In [30]:
#printing the test ys
y_test

585    5.292545
29     2.785439
270    2.713362
100    2.763425
278    2.497720
         ...   
168    3.384544
575    2.477143
563    2.072432
167    2.219519
285    2.100309
Name: percentage, Length: 118, dtype: float64

In [31]:
#calculating the mean squared error for linear regression
mse1 = mean_squared_error(y_test, y_predreg)

In [32]:
#MSE for Linear Regression model
mse1

0.3755572893890833

## ARIMA

In [35]:
#installing arima package
!pip install pmdarima

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.1/2.1 MB 7.1 MB/s eta 0:00:00


In [36]:
#installing libraries for arima
from statsmodels.tsa.arima.model import ARIMA
from sklearn.preprocessing import OneHotEncoder
from sklearn.compose import ColumnTransformer
from pmdarima import auto_arima

In [37]:
states

array(['Alaska', 'Alabama', 'Arkansas', 'Arizona', 'California',
       'Colorado', 'Connecticut', 'Florida', 'Georgia', 'Hawaii', 'Iowa',
       'Idaho', 'Illinois', 'Indiana', 'Kansas', 'Kentucky', 'Louisiana',
       'Massachusetts', 'Maryland', 'Maine', 'Michigan', 'Minnesota',
       'Missouri', 'Mississippi', 'Montana', 'North Carolina',
       'North Dakota', 'Nebraska', 'New Hampshire', 'New Jersey',
       'New Mexico', 'Nevada', 'New York', 'Ohio', 'Oklahoma', 'Oregon',
       'Pennsylvania', 'Rhode Island', 'South Carolina', 'South Dakota',
       'Tennessee', 'Texas', 'Utah', 'Virginia', 'Vermont', 'Washington',
       'Wisconsin', 'West Virginia', 'Wyoming'], dtype=object)

In [38]:
#ARIMA models for each state
mse_values = [] #to store mse for each state

# Splitting the data into train and test sets for each state
for state_label, state_name in zip(label_encoder.transform(states), states):
    state_data = weather[weather['state_encoded'] == state_label]
    X_train, X_test, y_train, y_test = train_test_split(state_data.drop(columns=['state', 'state_encoded']), state_data['percentage'], test_size=0.2, shuffle=False)

    # Checking if there's sufficient data for training
    if len(X_train) == 0 or len(X_test) == 0:
        print(f"Not enough data for state {state_name}. Skipping...")
        continue

    # Fitting ARIMA model on train data
    arima_model = auto_arima(y_train, seasonal=False, trace=True, error_action='ignore', suppress_warnings=True)

    # predicting for test data
    y_pred = arima_model.predict(n_periods=len(y_test))

    y_test_aligned = y_test.reset_index(drop=True)

    # Calculating Mean Squared Error
    mse = mean_squared_error(y_test_aligned, y_pred)
    mse_values.append((state_name, mse)) #appending to the index
#     print("\nypred are",y_pred,"yactual are",y_test_aligned,"\n")
#     print(f"\nState {state_name} - Mean Squared Error: {mse}\n")


Performing stepwise search to minimize aic
 ARIMA(1,0,1)(0,0,0)[0]             : AIC=16.383, Time=0.11 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=55.572, Time=0.01 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.04 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.07 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=18.663, Time=0.09 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=17.299, Time=0.21 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.17 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.17 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=inf, Time=0.41 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=9.857, Time=0.08 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=8.316, Time=0.05 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=10.859, Time=0.03 sec
 ARIMA(0,0,2)(0,0,0)[0] intercept   : AIC=8.599, Time=0.30 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=10.182, Time=0.11 sec
 ARIMA(1,0,2)(0,0,0)[0] intercept   : AIC=inf, Time=1.05 sec

Best model:  ARIM

/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(1,0,1)(0,0,0)[0]             : AIC=5.202, Time=0.30 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=44.567, Time=0.03 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.11 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.11 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=2.723, Time=0.48 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.19 sec
 ARIMA(3,0,1)(0,0,0)[0]             : AIC=inf, Time=0.44 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=8.050, Time=0.62 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=inf, Time=0.45 sec
 ARIMA(3,0,0)(0,0,0)[0]             : AIC=inf, Time=0.26 sec
 ARIMA(3,0,2)(0,0,0)[0]             : AIC=4.705, Time=0.88 sec
 ARIMA(2,0,1)(0,0,0)[0] intercept   : AIC=-6.594, Time=0.35 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=inf, Time=0.96 sec
 ARIMA(2,0,0)(0,0,0)[0] intercept   : AIC=-8.597, Time=0.45 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=-9.997, Time=0.15 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=-5.749, Time=0.03 sec
 

/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(1,0,1)(0,0,0)[0]             : AIC=1.227, Time=0.41 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=51.245, Time=0.03 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.11 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.29 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=0.769, Time=0.69 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.47 sec
 ARIMA(3,0,1)(0,0,0)[0]             : AIC=2.179, Time=0.76 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=2.583, Time=1.07 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=inf, Time=1.05 sec
 ARIMA(3,0,0)(0,0,0)[0]             : AIC=inf, Time=1.05 sec
 ARIMA(3,0,2)(0,0,0)[0]             : AIC=4.510, Time=1.05 sec
 ARIMA(2,0,1)(0,0,0)[0] intercept   : AIC=-4.157, Time=0.83 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=-3.861, Time=0.84 sec
 ARIMA(2,0,0)(0,0,0)[0] intercept   : AIC=-4.523, Time=0.59 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=-5.507, Time=0.23 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=-3.599, Time=0.03 

/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


Performing stepwise search to minimize aic
 ARIMA(1,0,1)(0,0,0)[0]             : AIC=inf, Time=0.32 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=32.378, Time=0.03 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.11 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.17 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=-26.867, Time=0.04 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=-26.820, Time=0.10 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=-26.905, Time=0.45 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=inf, Time=0.59 sec
 ARIMA(0,0,2)(0,0,0)[0] intercept   : AIC=inf, Time=0.67 sec
 ARIMA(1,0,2)(0,0,0)[0] intercept   : AIC=inf, Time=0.77 sec

Best model:  ARIMA(0,0,1)(0,0,0)[0] intercept
Total fit time: 3.304 seconds
Performing stepwise search to minimize aic


/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(1,0,1)(0,0,0)[0]             : AIC=1.005, Time=0.26 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=52.859, Time=0.02 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.07 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.20 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=inf, Time=0.62 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=3.335, Time=0.65 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.31 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.25 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=4.856, Time=1.17 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=inf, Time=0.85 sec

Best model:  ARIMA(1,0,1)(0,0,0)[0]          
Total fit time: 4.458 seconds
Performing stepwise search to minimize aic


/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(1,0,1)(0,0,0)[0]             : AIC=4.002, Time=0.62 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=42.344, Time=0.02 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.12 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.49 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=inf, Time=0.32 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=2.928, Time=0.51 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.25 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=inf, Time=0.76 sec
 ARIMA(1,0,3)(0,0,0)[0]             : AIC=inf, Time=0.64 sec
 ARIMA(0,0,3)(0,0,0)[0]             : AIC=inf, Time=0.39 sec
 ARIMA(2,0,3)(0,0,0)[0]             : AIC=inf, Time=1.02 sec
 ARIMA(1,0,2)(0,0,0)[0] intercept   : AIC=-6.228, Time=0.80 sec
 ARIMA(0,0,2)(0,0,0)[0] intercept   : AIC=inf, Time=0.50 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=inf, Time=0.71 sec
 ARIMA(2,0,2)(0,0,0)[0] intercept   : AIC=inf, Time=0.89 sec
 ARIMA(1,0,3)(0,0,0)[0] intercept   : AIC=inf, Time=0.95 sec
 ARIMA(0,0,1)(

/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(1,0,1)(0,0,0)[0]             : AIC=-0.287, Time=0.20 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=46.088, Time=0.01 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.04 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.07 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=1.909, Time=0.21 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=6.267, Time=0.37 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.27 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.21 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=4.437, Time=0.44 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=-7.365, Time=0.35 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=-9.365, Time=0.11 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=-9.378, Time=0.07 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=-8.817, Time=0.11 sec

Best model:  ARIMA(0,0,0)(0,0,0)[0] intercept
Total fit time: 2.503 seconds
Performing stepwise search to minimize aic


/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(1,0,1)(0,0,0)[0]             : AIC=1.880, Time=0.33 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=46.007, Time=0.03 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.05 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.10 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=inf, Time=0.42 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=2.982, Time=0.44 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.08 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.07 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=inf, Time=0.22 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=inf, Time=0.21 sec

Best model:  ARIMA(1,0,1)(0,0,0)[0]          
Total fit time: 1.958 seconds
Performing stepwise search to minimize aic
 ARIMA(1,0,1)(0,0,0)[0]             : AIC=15.679, Time=0.08 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=52.535, Time=0.01 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.03 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.05 sec


/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(2,0,1)(0,0,0)[0]             : AIC=15.029, Time=0.14 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.07 sec
 ARIMA(3,0,1)(0,0,0)[0]             : AIC=inf, Time=0.19 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=12.822, Time=0.20 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=14.397, Time=0.16 sec
 ARIMA(3,0,2)(0,0,0)[0]             : AIC=15.108, Time=0.27 sec
 ARIMA(2,0,3)(0,0,0)[0]             : AIC=inf, Time=0.16 sec
 ARIMA(1,0,3)(0,0,0)[0]             : AIC=inf, Time=0.13 sec
 ARIMA(3,0,3)(0,0,0)[0]             : AIC=15.727, Time=0.25 sec
 ARIMA(2,0,2)(0,0,0)[0] intercept   : AIC=9.133, Time=0.24 sec
 ARIMA(1,0,2)(0,0,0)[0] intercept   : AIC=inf, Time=0.22 sec
 ARIMA(2,0,1)(0,0,0)[0] intercept   : AIC=12.075, Time=0.22 sec
 ARIMA(3,0,2)(0,0,0)[0] intercept   : AIC=10.888, Time=0.31 sec
 ARIMA(2,0,3)(0,0,0)[0] intercept   : AIC=10.865, Time=0.24 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=10.042, Time=0.21 sec
 ARIMA(1,0,3)(0,0,0)[0] intercept   : AIC=inf, Time=0.24

/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.06 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=4.170, Time=0.08 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=4.179, Time=0.19 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.09 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.13 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=inf, Time=0.23 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=-4.878, Time=0.12 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=-6.839, Time=0.05 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=-7.562, Time=0.03 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=-6.781, Time=0.05 sec

Best model:  ARIMA(0,0,0)(0,0,0)[0] intercept
Total fit time: 1.188 seconds
Performing stepwise search to minimize aic
 ARIMA(1,0,1)(0,0,0)[0]             : AIC=12.900, Time=0.06 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=53.005, Time=0.01 sec


/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.06 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.06 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=inf, Time=0.19 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=14.879, Time=0.16 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.10 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.07 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=inf, Time=0.20 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=9.593, Time=0.14 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=9.145, Time=0.04 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=9.578, Time=0.04 sec
 ARIMA(0,0,2)(0,0,0)[0] intercept   : AIC=10.597, Time=0.14 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=7.676, Time=0.07 sec
 ARIMA(2,0,0)(0,0,0)[0] intercept   : AIC=9.596, Time=0.08 sec
 ARIMA(2,0,1)(0,0,0)[0] intercept   : AIC=11.600, Time=0.19 sec

Best model:  ARIMA(1,0,0)(0,0,0)[0] intercept
Total fit time: 1.651 seconds
Performing stepwise search to minimize aic
 ARIMA(

/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.08 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=-9.696, Time=0.17 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=inf, Time=0.19 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.09 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.10 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=-9.305, Time=0.21 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=-18.135, Time=0.16 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=-20.133, Time=0.05 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=-22.121, Time=0.03 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=-20.138, Time=0.05 sec

Best model:  ARIMA(0,0,0)(0,0,0)[0] intercept
Total fit time: 1.306 seconds
Performing stepwise search to minimize aic


/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(1,0,1)(0,0,0)[0]             : AIC=-10.226, Time=0.17 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=41.159, Time=0.01 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.06 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.04 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=inf, Time=0.14 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=-8.597, Time=0.17 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.10 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.13 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=-6.234, Time=0.21 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=-15.832, Time=0.25 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=-16.756, Time=0.42 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=-12.928, Time=0.08 sec
 ARIMA(0,0,2)(0,0,0)[0] intercept   : AIC=-15.686, Time=0.28 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=-17.190, Time=0.19 sec
 ARIMA(2,0,0)(0,0,0)[0] intercept   : AIC=-15.514, Time=0.29 sec
 ARIMA(2,0,1)(0,0,0)[0] intercept   : AIC=-13.92

/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(1,0,1)(0,0,0)[0]             : AIC=6.353, Time=0.35 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=48.420, Time=0.03 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.08 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.19 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=inf, Time=0.45 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=inf, Time=0.17 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.08 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.11 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=inf, Time=0.19 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=-2.751, Time=0.16 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=inf, Time=0.11 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=-3.701, Time=0.04 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=-4.937, Time=0.03 sec

Best model:  ARIMA(0,0,0)(0,0,0)[0] intercept
Total fit time: 2.030 seconds
Performing stepwise search to minimize aic
 ARIMA(1,0,1)(0,0,0)[0]             : AIC=1.902, Time=0.09 sec


/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(0,0,0)(0,0,0)[0]             : AIC=44.192, Time=0.02 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.04 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.10 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=0.987, Time=0.16 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.06 sec
 ARIMA(3,0,1)(0,0,0)[0]             : AIC=2.982, Time=0.17 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=3.140, Time=0.20 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=5.355, Time=0.16 sec
 ARIMA(3,0,0)(0,0,0)[0]             : AIC=inf, Time=0.08 sec
 ARIMA(3,0,2)(0,0,0)[0]             : AIC=4.924, Time=0.27 sec
 ARIMA(2,0,1)(0,0,0)[0] intercept   : AIC=-4.763, Time=0.17 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=-5.367, Time=0.16 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=-6.308, Time=0.04 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=-8.287, Time=0.03 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=-6.319, Time=0.05 sec

Best model:  ARIMA(0,0,0)(0,0,0)[0] intercept
Total fit 

/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(1,0,1)(0,0,0)[0]             : AIC=3.631, Time=0.07 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=39.895, Time=0.02 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.04 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.07 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=inf, Time=0.15 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=7.277, Time=0.17 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.08 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.07 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=7.674, Time=0.19 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=-1.566, Time=0.08 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=-3.204, Time=0.05 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=0.141, Time=0.03 sec
 ARIMA(0,0,2)(0,0,0)[0] intercept   : AIC=inf, Time=0.18 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=-2.369, Time=0.04 sec
 ARIMA(1,0,2)(0,0,0)[0] intercept   : AIC=-0.382, Time=0.24 sec

Best model:  ARIMA(0,0,1)(0,0,0)[0] intercept
Total fit time:

/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.07 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=-6.203, Time=0.16 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.08 sec
 ARIMA(3,0,1)(0,0,0)[0]             : AIC=-4.154, Time=0.18 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=-2.553, Time=0.21 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=inf, Time=0.20 sec
 ARIMA(3,0,0)(0,0,0)[0]             : AIC=inf, Time=0.10 sec
 ARIMA(3,0,2)(0,0,0)[0]             : AIC=-1.199, Time=0.23 sec
 ARIMA(2,0,1)(0,0,0)[0] intercept   : AIC=-15.695, Time=0.15 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=-19.521, Time=0.27 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=inf, Time=0.09 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=-19.566, Time=0.10 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=-17.203, Time=0.04 sec
 ARIMA(2,0,0)(0,0,0)[0] intercept   : AIC=-17.756, Time=0.12 sec

Best model:  ARIMA(1,0,0)(0,0,0)[0] intercept
Total fit time: 2.192 seconds
Performing stepwise search to minimiz

/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(1,0,1)(0,0,0)[0]             : AIC=-3.912, Time=0.13 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=45.815, Time=0.01 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.04 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.05 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=-4.334, Time=0.17 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.07 sec
 ARIMA(3,0,1)(0,0,0)[0]             : AIC=-3.993, Time=0.24 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=inf, Time=0.22 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=inf, Time=0.13 sec
 ARIMA(3,0,0)(0,0,0)[0]             : AIC=inf, Time=0.15 sec
 ARIMA(3,0,2)(0,0,0)[0]             : AIC=-2.516, Time=0.25 sec
 ARIMA(2,0,1)(0,0,0)[0] intercept   : AIC=-16.318, Time=0.17 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=inf, Time=0.15 sec
 ARIMA(2,0,0)(0,0,0)[0] intercept   : AIC=-17.135, Time=0.13 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=-17.976, Time=0.07 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=-16.162, Time=0.

/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(1,0,1)(0,0,0)[0]             : AIC=17.683, Time=0.08 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=44.584, Time=0.01 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=19.990, Time=0.04 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.05 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=18.750, Time=0.13 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=inf, Time=0.13 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.09 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.06 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=inf, Time=0.20 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=11.679, Time=0.12 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=9.838, Time=0.04 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=8.425, Time=0.03 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=9.694, Time=0.05 sec

Best model:  ARIMA(0,0,0)(0,0,0)[0] intercept
Total fit time: 1.036 seconds
Performing stepwise search to minimize aic
 ARIMA(1,0,1)(0,0,0)[0]             : AIC=-2.535, Time=0.11 sec
 A

/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.06 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=inf, Time=0.38 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=-3.087, Time=0.46 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.24 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=inf, Time=0.44 sec
 ARIMA(1,0,3)(0,0,0)[0]             : AIC=inf, Time=0.36 sec
 ARIMA(0,0,3)(0,0,0)[0]             : AIC=inf, Time=0.54 sec
 ARIMA(2,0,3)(0,0,0)[0]             : AIC=inf, Time=0.61 sec
 ARIMA(1,0,2)(0,0,0)[0] intercept   : AIC=-7.156, Time=0.65 sec
 ARIMA(0,0,2)(0,0,0)[0] intercept   : AIC=inf, Time=0.21 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=-7.075, Time=0.18 sec
 ARIMA(2,0,2)(0,0,0)[0] intercept   : AIC=-4.501, Time=0.26 sec
 ARIMA(1,0,3)(0,0,0)[0] intercept   : AIC=-5.428, Time=0.26 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=-8.675, Time=0.11 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=-10.353, Time=0.03 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=-8.930, Time=0.10 sec

/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.08 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=1.227, Time=0.16 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=0.938, Time=0.17 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.09 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.05 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=3.190, Time=0.18 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=-6.171, Time=0.24 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=-8.101, Time=0.06 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=-9.873, Time=0.02 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=-8.166, Time=0.07 sec

Best model:  ARIMA(0,0,0)(0,0,0)[0] intercept
Total fit time: 1.265 seconds
Performing stepwise search to minimize aic


/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(1,0,1)(0,0,0)[0]             : AIC=-0.134, Time=0.12 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=44.925, Time=0.01 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.06 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.05 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=inf, Time=0.12 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=inf, Time=0.18 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.11 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.12 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=3.779, Time=0.20 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=-9.156, Time=0.22 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=-10.908, Time=0.06 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=-11.784, Time=0.03 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=-9.881, Time=0.07 sec

Best model:  ARIMA(0,0,0)(0,0,0)[0] intercept
Total fit time: 1.368 seconds
Performing stepwise search to minimize aic


/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(1,0,1)(0,0,0)[0]             : AIC=inf, Time=0.15 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=43.223, Time=0.01 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.04 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.05 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=-11.070, Time=0.03 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=-12.380, Time=0.18 sec
 ARIMA(2,0,0)(0,0,0)[0] intercept   : AIC=-10.540, Time=0.10 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=-10.570, Time=0.21 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=inf, Time=0.09 sec
 ARIMA(2,0,1)(0,0,0)[0] intercept   : AIC=-8.518, Time=0.17 sec

Best model:  ARIMA(1,0,0)(0,0,0)[0] intercept
Total fit time: 1.043 seconds
Performing stepwise search to minimize aic
 ARIMA(1,0,1)(0,0,0)[0]             : AIC=16.396, Time=0.11 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=51.526, Time=0.01 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.03 sec


/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.07 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=18.323, Time=0.17 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=18.332, Time=0.18 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.07 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.06 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=inf, Time=0.22 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=inf, Time=0.14 sec

Best model:  ARIMA(1,0,1)(0,0,0)[0]          
Total fit time: 1.077 seconds
Performing stepwise search to minimize aic


/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(1,0,1)(0,0,0)[0]             : AIC=0.516, Time=0.09 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=47.284, Time=0.01 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.05 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.06 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=inf, Time=0.15 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=2.192, Time=0.22 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.10 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.09 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=4.483, Time=0.20 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=-4.910, Time=0.12 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=-5.201, Time=0.05 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=-1.826, Time=0.02 sec
 ARIMA(0,0,2)(0,0,0)[0] intercept   : AIC=-4.555, Time=0.17 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=-6.262, Time=0.06 sec
 ARIMA(2,0,0)(0,0,0)[0] intercept   : AIC=-4.964, Time=0.11 sec
 ARIMA(2,0,1)(0,0,0)[0] intercept   : AIC=-2.673, Time=0.2

/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.07 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=30.245, Time=0.06 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=30.096, Time=0.17 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.08 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.07 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=32.188, Time=0.19 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=22.543, Time=0.15 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=20.701, Time=0.03 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=18.721, Time=0.04 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=20.696, Time=0.05 sec

Best model:  ARIMA(0,0,0)(0,0,0)[0] intercept
Total fit time: 1.085 seconds
Performing stepwise search to minimize aic
 ARIMA(1,0,1)(0,0,0)[0]             : AIC=8.447, Time=0.12 sec


/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(0,0,0)(0,0,0)[0]             : AIC=45.745, Time=0.01 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.04 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.07 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=11.011, Time=0.08 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=inf, Time=0.17 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.12 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.07 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=inf, Time=0.20 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=inf, Time=0.16 sec

Best model:  ARIMA(1,0,1)(0,0,0)[0]          
Total fit time: 1.044 seconds
Performing stepwise search to minimize aic


/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(1,0,1)(0,0,0)[0]             : AIC=22.323, Time=0.15 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=50.522, Time=0.01 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=22.120, Time=0.05 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.15 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.10 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=24.323, Time=0.23 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=16.060, Time=0.09 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=14.578, Time=0.04 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=15.781, Time=0.09 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=inf, Time=0.32 sec

Best model:  ARIMA(0,0,0)(0,0,0)[0] intercept
Total fit time: 1.251 seconds
Performing stepwise search to minimize aic


/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(1,0,1)(0,0,0)[0]             : AIC=-3.681, Time=0.21 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=36.109, Time=0.04 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.10 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.14 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=-3.688, Time=0.39 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.28 sec
 ARIMA(3,0,1)(0,0,0)[0]             : AIC=-2.963, Time=0.60 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=-2.711, Time=0.31 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=-4.076, Time=0.15 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.16 sec
 ARIMA(1,0,3)(0,0,0)[0]             : AIC=inf, Time=0.14 sec
 ARIMA(0,0,3)(0,0,0)[0]             : AIC=inf, Time=0.16 sec
 ARIMA(2,0,3)(0,0,0)[0]             : AIC=-0.447, Time=0.23 sec
 ARIMA(1,0,2)(0,0,0)[0] intercept   : AIC=-8.705, Time=0.20 sec
 ARIMA(0,0,2)(0,0,0)[0] intercept   : AIC=-8.730, Time=0.22 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=inf, Time=0.10 s

/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(0,0,1)(0,0,0)[0]             : AIC=41.730, Time=0.06 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=21.441, Time=0.18 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=18.700, Time=0.17 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.09 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.05 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=21.460, Time=0.19 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=11.582, Time=0.14 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=inf, Time=0.10 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=9.679, Time=0.05 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=8.207, Time=0.03 sec

Best model:  ARIMA(0,0,0)(0,0,0)[0] intercept
Total fit time: 1.238 seconds
Performing stepwise search to minimize aic
 ARIMA(1,0,1)(0,0,0)[0]             : AIC=17.355, Time=0.05 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=54.210, Time=0.01 sec


/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.04 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.10 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=19.065, Time=0.08 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=inf, Time=0.15 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.08 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.06 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=21.304, Time=0.20 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=11.360, Time=0.16 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=9.392, Time=0.05 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=7.677, Time=0.03 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=9.360, Time=0.05 sec

Best model:  ARIMA(0,0,0)(0,0,0)[0] intercept
Total fit time: 1.080 seconds
Performing stepwise search to minimize aic
 ARIMA(1,0,1)(0,0,0)[0]             : AIC=-11.252, Time=0.08 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=33.481, Time=0.02 sec


/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.06 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.07 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=-9.301, Time=0.14 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=-9.265, Time=0.15 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.10 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.08 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=-5.695, Time=0.21 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=-18.544, Time=0.07 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=-20.564, Time=0.04 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=-22.543, Time=0.03 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=-20.565, Time=0.09 sec

Best model:  ARIMA(0,0,0)(0,0,0)[0] intercept
Total fit time: 1.172 seconds
Performing stepwise search to minimize aic
 ARIMA(1,0,1)(0,0,0)[0]             : AIC=inf, Time=0.10 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=37.019, Time=0.01 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.05 se

/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.08 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=-17.812, Time=0.03 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=-15.906, Time=0.08 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=inf, Time=0.16 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=inf, Time=0.15 sec

Best model:  ARIMA(0,0,0)(0,0,0)[0] intercept
Total fit time: 0.669 seconds
Performing stepwise search to minimize aic


/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(1,0,1)(0,0,0)[0]             : AIC=inf, Time=0.09 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=46.256, Time=0.01 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.05 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.07 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=-10.333, Time=0.03 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=-12.477, Time=0.09 sec
 ARIMA(2,0,0)(0,0,0)[0] intercept   : AIC=-12.746, Time=0.14 sec
 ARIMA(3,0,0)(0,0,0)[0] intercept   : AIC=-8.490, Time=0.27 sec
 ARIMA(2,0,1)(0,0,0)[0] intercept   : AIC=-10.951, Time=0.23 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=-13.034, Time=0.10 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=inf, Time=0.12 sec
 ARIMA(1,0,2)(0,0,0)[0] intercept   : AIC=-11.162, Time=0.23 sec
 ARIMA(0,0,2)(0,0,0)[0] intercept   : AIC=-13.131, Time=0.12 sec
 ARIMA(0,0,3)(0,0,0)[0] intercept   : AIC=-11.162, Time=0.17 sec
 ARIMA(1,0,3)(0,0,0)[0] intercept   : AIC=-9.808, Time=0.33 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=in

/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(1,0,1)(0,0,0)[0]             : AIC=11.176, Time=0.12 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=49.761, Time=0.02 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.05 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.10 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=10.407, Time=0.13 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.07 sec
 ARIMA(3,0,1)(0,0,0)[0]             : AIC=11.854, Time=0.19 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=11.844, Time=0.21 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=11.528, Time=0.15 sec
 ARIMA(3,0,0)(0,0,0)[0]             : AIC=inf, Time=0.10 sec
 ARIMA(3,0,2)(0,0,0)[0]             : AIC=13.830, Time=0.27 sec
 ARIMA(2,0,1)(0,0,0)[0] intercept   : AIC=4.005, Time=0.64 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=3.608, Time=0.46 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=3.346, Time=0.15 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=1.351, Time=0.07 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=3.342, Time=0.

/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(1,0,1)(0,0,0)[0]             : AIC=-6.569, Time=0.31 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=38.946, Time=0.02 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.09 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.14 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=inf, Time=0.38 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=-5.562, Time=0.43 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.24 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.25 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=-2.710, Time=0.25 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=-12.420, Time=0.16 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=-13.470, Time=0.05 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=-12.740, Time=0.03 sec
 ARIMA(0,0,2)(0,0,0)[0] intercept   : AIC=inf, Time=0.14 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=-14.410, Time=0.09 sec
 ARIMA(2,0,0)(0,0,0)[0] intercept   : AIC=-12.428, Time=0.12 sec
 ARIMA(2,0,1)(0,0,0)[0] intercept   : AIC=-10.410, Ti

/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.06 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=17.072, Time=0.13 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=inf, Time=0.16 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.10 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.06 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=inf, Time=0.21 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=11.100, Time=0.15 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=9.169, Time=0.05 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=7.193, Time=0.04 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=9.161, Time=0.04 sec

Best model:  ARIMA(0,0,0)(0,0,0)[0] intercept
Total fit time: 1.161 seconds
Performing stepwise search to minimize aic
 ARIMA(1,0,1)(0,0,0)[0]             : AIC=inf, Time=0.12 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=49.981, Time=0.01 sec


/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.06 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.08 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=-7.969, Time=0.03 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=-5.969, Time=0.06 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=inf, Time=0.11 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=inf, Time=0.21 sec

Best model:  ARIMA(0,0,0)(0,0,0)[0] intercept
Total fit time: 0.696 seconds
Performing stepwise search to minimize aic
 ARIMA(1,0,1)(0,0,0)[0]             : AIC=inf, Time=0.11 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=49.164, Time=0.01 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.03 sec


/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.07 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=2.310, Time=0.03 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=3.055, Time=0.04 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=inf, Time=0.08 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=2.379, Time=0.16 sec

Best model:  ARIMA(0,0,0)(0,0,0)[0] intercept
Total fit time: 0.553 seconds
Performing stepwise search to minimize aic
 ARIMA(1,0,1)(0,0,0)[0]             : AIC=8.317, Time=0.10 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=46.290, Time=0.01 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.06 sec


/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.07 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=10.314, Time=0.10 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=10.312, Time=0.20 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.10 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.08 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=12.316, Time=0.20 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=2.516, Time=0.06 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=0.552, Time=0.04 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=-1.447, Time=0.03 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=0.553, Time=0.07 sec

Best model:  ARIMA(0,0,0)(0,0,0)[0] intercept
Total fit time: 1.128 seconds


/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


Performing stepwise search to minimize aic
 ARIMA(1,0,1)(0,0,0)[0]             : AIC=-8.445, Time=0.08 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=40.098, Time=0.01 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.06 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.07 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=-6.525, Time=0.15 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=inf, Time=0.16 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.10 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.07 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=-4.453, Time=0.20 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=inf, Time=0.17 sec

Best model:  ARIMA(1,0,1)(0,0,0)[0]          
Total fit time: 1.074 seconds
Performing stepwise search to minimize aic


/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(1,0,1)(0,0,0)[0]             : AIC=9.375, Time=0.09 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=48.750, Time=0.01 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.04 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.06 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=inf, Time=0.17 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=11.358, Time=0.17 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.12 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.09 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=13.108, Time=0.21 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=3.964, Time=0.18 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=1.989, Time=0.04 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=0.443, Time=0.03 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=1.964, Time=0.08 sec

Best model:  ARIMA(0,0,0)(0,0,0)[0] intercept
Total fit time: 1.317 seconds
Performing stepwise search to minimize aic


/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(1,0,1)(0,0,0)[0]             : AIC=0.061, Time=0.15 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=48.640, Time=0.01 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.04 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.05 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=inf, Time=0.14 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=3.303, Time=0.17 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.09 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.07 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=inf, Time=0.21 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=inf, Time=0.22 sec

Best model:  ARIMA(1,0,1)(0,0,0)[0]          
Total fit time: 1.172 seconds
Performing stepwise search to minimize aic
 ARIMA(1,0,1)(0,0,0)[0]             : AIC=16.439, Time=0.12 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=50.982, Time=0.01 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.04 sec


/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.07 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=inf, Time=0.14 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=19.941, Time=0.28 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.07 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.06 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=inf, Time=0.23 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=inf, Time=0.14 sec

Best model:  ARIMA(1,0,1)(0,0,0)[0]          
Total fit time: 1.185 seconds
Performing stepwise search to minimize aic


/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(1,0,1)(0,0,0)[0]             : AIC=4.450, Time=0.12 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=48.930, Time=0.01 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.08 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.05 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=inf, Time=0.15 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=5.419, Time=0.15 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.14 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.21 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=8.531, Time=0.61 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=0.662, Time=0.23 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=1.214, Time=0.09 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=-0.935, Time=0.15 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=3.717, Time=0.03 sec
 ARIMA(2,0,0)(0,0,0)[0] intercept   : AIC=0.498, Time=0.26 sec
 ARIMA(2,0,1)(0,0,0)[0] intercept   : AIC=1.765, Time=0.55 sec

Best model:  ARIMA(1,0,0)(0,0,0)[0] intercept
Total fit time: 

/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(1,0,1)(0,0,0)[0]             : AIC=3.159, Time=0.21 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=39.193, Time=0.01 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=inf, Time=0.05 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.16 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=3.448, Time=0.50 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=2.166, Time=0.23 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.10 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=inf, Time=0.20 sec
 ARIMA(1,0,3)(0,0,0)[0]             : AIC=inf, Time=0.17 sec
 ARIMA(0,0,3)(0,0,0)[0]             : AIC=inf, Time=0.14 sec
 ARIMA(2,0,3)(0,0,0)[0]             : AIC=inf, Time=0.23 sec
 ARIMA(1,0,2)(0,0,0)[0] intercept   : AIC=-4.385, Time=0.19 sec
 ARIMA(0,0,2)(0,0,0)[0] intercept   : AIC=inf, Time=0.52 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=-2.034, Time=0.41 sec
 ARIMA(2,0,2)(0,0,0)[0] intercept   : AIC=-2.331, Time=0.22 sec
 ARIMA(1,0,3)(0,0,0)[0] intercept   : AIC=inf, Time=0.22 sec
 ARIMA

/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.08 sec
 ARIMA(2,0,1)(0,0,0)[0]             : AIC=13.613, Time=0.08 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=13.659, Time=0.17 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.09 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=inf, Time=0.06 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=14.968, Time=0.22 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=5.605, Time=0.11 sec
 ARIMA(0,0,1)(0,0,0)[0] intercept   : AIC=3.791, Time=0.04 sec
 ARIMA(0,0,0)(0,0,0)[0] intercept   : AIC=1.992, Time=0.03 sec
 ARIMA(1,0,0)(0,0,0)[0] intercept   : AIC=3.738, Time=0.11 sec

Best model:  ARIMA(0,0,0)(0,0,0)[0] intercept
Total fit time: 1.154 seconds
Performing stepwise search to minimize aic
 ARIMA(1,0,1)(0,0,0)[0]             : AIC=33.799, Time=0.05 sec
 ARIMA(0,0,0)(0,0,0)[0]             : AIC=56.313, Time=0.01 sec
 ARIMA(1,0,0)(0,0,0)[0]             : AIC=34.281, Time=0.03 sec
 ARIMA(0,0,1)(0,0,0)[0]             : AIC=inf, Time=0.05 sec


/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


 ARIMA(2,0,1)(0,0,0)[0]             : AIC=inf, Time=0.12 sec
 ARIMA(1,0,2)(0,0,0)[0]             : AIC=inf, Time=0.15 sec
 ARIMA(0,0,2)(0,0,0)[0]             : AIC=inf, Time=0.09 sec
 ARIMA(2,0,0)(0,0,0)[0]             : AIC=35.648, Time=0.06 sec
 ARIMA(2,0,2)(0,0,0)[0]             : AIC=inf, Time=0.18 sec
 ARIMA(1,0,1)(0,0,0)[0] intercept   : AIC=inf, Time=0.15 sec

Best model:  ARIMA(1,0,1)(0,0,0)[0]          
Total fit time: 0.905 seconds


/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/usr/local/lib/python3.10/dist-packages/statsmodels/tsa/base/tsa_model.py:836: FutureWarning: No supported index is available. In the next version, calling this method in a model without a supported index will result in an exception.
  return get_prediction_index(


In [39]:
#creating a dataframe of the mse array
MSEstate = pd.DataFrame(mse_values, columns=['State', 'MSE'])

In [40]:
#statistics of mse for all states
MSEstate.describe()

MSE
count  49.000000
mean    0.131770
std     0.229311
min     0.004619
25%     0.024314
50%     0.049143
75%     0.131105
max     1.381454

In [41]:
MSEstate['MSE'].median()

0.04914265101050422

## Least Squares Dummy Variable Regression (LSDV)

In [42]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2,random_state=1234)

In [43]:
#test-train-random split is better
import statsmodels.api as sm

# Adding a constant term to the features matrix (required for the intercept term)
X_lsdv = sm.add_constant(X_train)

# Fitting model
model_lsdv = sm.OLS(y_train, X_lsdv)
result_lsdv = model_lsdv.fit()

# Predicting on test set
X_test_lsdv = sm.add_constant(X_test)
y_pred_lsdv = result_lsdv.predict(X_test_lsdv)

# Calculating MSE
mse_lsdv = mean_squared_error(y_test, y_pred_lsdv)

In [44]:
#MSE for LSDV model
mse_lsdv

0.3755572893787338

In [45]:
pip install tensorflow

## Neural Network

In [46]:
#random splitting test and train data
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2,random_state=1234)

In [47]:
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense

# Defining the neural network model
model_nn = Sequential([
    Dense(64, activation='relu', input_shape=(X_train.shape[1],)),
    Dense(32, activation='relu'),
    Dense(1)  # Output layer with 1 neuron for regression
])

# Compiling the model
model_nn.compile(optimizer='adam', loss='mean_squared_error')

# Training model
model_nn.fit(X_train, y_train, epochs=50, batch_size=32, verbose=0)

# Predicting on test set
y_pred_nn = model_nn.predict(X_test).flatten()  # Flatten to convert to 1D array

# Calculating MSE
mse_nn = mean_squared_error(y_test, y_pred_nn)


4/4 [==============================] - 0s 4ms/step


In [48]:
#Data loss
mse_nn

86502.34578228462

In [49]:
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense, Dropout
from tensorflow.keras.optimizers import Adam
from tensorflow.keras.callbacks import EarlyStopping
from sklearn.metrics import mean_squared_error
from sklearn.preprocessing import StandardScaler

# Feature scaling
scaler = StandardScaler()
X_train_scaled = scaler.fit_transform(X_train)
X_test_scaled = scaler.transform(X_test)

# Defining the neural network mode
model_nn = Sequential([
    Dense(64, activation='relu', input_shape=(X_train_scaled.shape[1],)),
    Dropout(0.2),  # Dropout layer to prevent overfitting
    Dense(32, activation='relu'),
    Dropout(0.2),
    Dense(1)  # Output layer with 1 neuron for regression
])

# Compiling the model
model_nn.compile(optimizer=Adam(learning_rate=0.001), loss='mean_squared_error')

# Defining early stopping criteria
early_stopping = EarlyStopping(monitor='val_loss', patience=5, restore_best_weights=True)

# Training the model with early stopping
history = model_nn.fit(X_train_scaled, y_train,
                        validation_split=0.2,  # Use a portion of training data for validation
                        epochs=100,
                        batch_size=32,
                        callbacks=[early_stopping],
                        verbose=1)

# Predicting on the test set
y_pred_nn = model_nn.predict(X_test_scaled).flatten()

# Calculating MSE
mse_nn = mean_squared_error(y_test, y_pred_nn)
mse_nn

Epoch 1/100
12/12 [==============================] - 2s 42ms/step - loss: 9.9764 - val_loss: 7.9305
Epoch 2/100
12/12 [==============================] - 0s 9ms/step - loss: 5.9631 - val_loss: 4.2959
Epoch 3/100
12/12 [==============================] - 0s 14ms/step - loss: 3.2923 - val_loss: 1.9696
Epoch 4/100
12/12 [==============================] - 0s 13ms/step - loss: 1.7903 - val_loss: 0.8889
Epoch 5/100
12/12 [==============================] - 0s 13ms/step - loss: 1.3447 - val_loss: 0.6801
Epoch 6/100
12/12 [==============================] - 0s 10ms/step - loss: 0.9852 - val_loss: 0.5825
Epoch 7/100
12/12 [==============================] - 0s 10ms/step - loss: 0.9432 - val_loss: 0.5157
Epoch 8/100
12/12 [==============================] - 0s 14ms/step - loss: 0.9540 - val_loss: 0.4948
Epoch 9/100
12/12 [==============================] - 0s 11ms/step - loss: 0.8354 - val_loss: 0.4816
Epoch 10/100
12/12 [==============================] - 0s 10ms/step - loss: 0.8383 - val_loss: 0.4614


0.30846181316333937

## CNN

In [50]:
train_data = weather[(weather['year'] > 2009) & (weather['year'] <= 2019)]
test_data = weather[weather['year'] > 2019]
#didn't split the data this way for others as the mse was higher

In [51]:
X_train = train_data.drop(['percentage', 'state'], axis=1)
y_train = train_data['percentage']

In [52]:
X_test = test_data.drop(['percentage', 'state'], axis=1)
y_test = test_data['percentage']

In [53]:
import numpy as np
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Conv1D, MaxPooling1D, Flatten, Dense

# Reshaping data for CNN input (assuming 1D time series)
X_train_reshaped = X_train.values.reshape(X_train.shape[0], X_train.shape[1], 1)
X_test_reshaped = X_test.values.reshape(X_test.shape[0], X_test.shape[1], 1)

# Defining the CNN model
model = Sequential([
    Conv1D(filters=64, kernel_size=3, activation='relu', input_shape=(X_train.shape[1], 1)),
    MaxPooling1D(pool_size=2),
    Conv1D(filters=32, kernel_size=3, activation='relu'),
    MaxPooling1D(pool_size=2),
    Flatten(),
    Dense(50, activation='relu'),
    Dense(1)  # Output layer (1 neuron for regression)
])

# Compiling the model
model.compile(optimizer='adam', loss='mse')

# Training the model
model.fit(X_train_reshaped, y_train, epochs=10, batch_size=32, validation_split=0.1)




Epoch 1/10
14/14 [==============================] - 3s 33ms/step - loss: 38222757888.0000 - val_loss: 1494652672.0000
Epoch 2/10
14/14 [==============================] - 0s 18ms/step - loss: 3474200576.0000 - val_loss: 3250160.7500
Epoch 3/10
14/14 [==============================] - 0s 13ms/step - loss: 724311360.0000 - val_loss: 53229368.0000
Epoch 4/10
14/14 [==============================] - 0s 14ms/step - loss: 93558664.0000 - val_loss: 9196822.0000
Epoch 5/10
14/14 [==============================] - 0s 12ms/step - loss: 11458955.0000 - val_loss: 708069.0625
Epoch 6/10
14/14 [==============================] - 0s 20ms/step - loss: 4453759.5000 - val_loss: 983876.8750
Epoch 7/10
14/14 [==============================] - 0s 9ms/step - loss: 1939142.3750 - val_loss: 110881.7891
Epoch 8/10
14/14 [==============================] - 0s 13ms/step - loss: 280994.6250 - val_loss: 3944.2549
Epoch 9/10
14/14 [==============================] - 0s 10ms/step - loss: 83481.7578 - val_loss: 11828.245

In [54]:
X_test.head()

year  yearly_avg_snow  yearly_max_temp  yearly_min_temp  yearly_avg_temp  \
10  2021         9.310833             75.4            -30.4        32.662500   
11  2022         6.554167             74.3            -23.1        35.154167   
22  2021         0.100000             91.3             31.9        65.120000   
23  2022         0.094167             95.3             28.4        65.145000   
34  2021         0.952500             93.7             20.9        62.807500   

    yearly_avg_prcp  yearly_min_prcp  yearly_max_prcp  total pop  \
10         3.027500             2.00             4.82     720077   
11         3.630000             0.89             6.97     719005   
22         5.516667             1.57             9.76    4977176   
23         4.730833             2.36             7.72    5008263   
34         3.911667             1.77             6.93    2984483   

    in-state pop  state_encoded  
10        688699              1  
11        682442              1  
22       4861535              0  
23       4869000              0  
34       2908375              3

In [55]:
#Predicting on test data
y_predcnn = model.predict(X_test_reshaped)
mse_cnn = mean_squared_error(y_test, y_predcnn)
mse_cnn # MSE for CNNmodel

4/4 [==============================] - 0s 6ms/step


62773.63033925313

## SVM

In [56]:
#random splitting the data
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2,random_state=1234)

In [57]:
#test-train-random split is better
from sklearn.svm import SVR
from sklearn.preprocessing import StandardScaler
from sklearn.pipeline import make_pipeline


# Defining the SVR model with a radial basis function (RBF) kernel
svm_model = make_pipeline(StandardScaler(), SVR(kernel='rbf', C=5, gamma=0.1, epsilon=0.02))

# Training SVR model
svm_model.fit(X_train, y_train)

# Predicting on test set
y_pred_svm = svm_model.predict(X_test)

# Calculating MSE
mse_svm = mean_squared_error(y_test, y_pred_svm)
mse_svm #MSE for SVM model


0.1597960748805673

In [58]:
#MAE
from sklearn.metrics import mean_absolute_error
mae = mean_absolute_error(y_test,y_pred_svm)
print(mae)

0.31048027718968196


## Desicion Tree

In [59]:
#random split for train-test data
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2,random_state=1234)

In [60]:
#creating instance of Decision tree
from sklearn import tree
myTree = tree.DecisionTreeRegressor(max_depth=2)

In [61]:
#fitting the desicion tree
#test-train-random split is better
myTree.fit(X_train,y_train)

DecisionTreeRegressor(max_depth=2)

In [62]:
#predicting for test data
y_preddt = myTree.predict(X_test)

In [63]:
#MSE for Desicion tree model
mse_dt = mean_squared_error(y_test, y_preddt)
mse_dt

0.5027920725953638

## GBM

In [64]:
#random split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2,random_state=1234)

In [65]:
from sklearn.ensemble import GradientBoostingRegressor

In [66]:
#creating instance of GBM
gbm = GradientBoostingRegressor()

In [67]:
#fitting GBM on train data
gbm.fit(X_train, y_train)

GradientBoostingRegressor()

In [68]:
#predicting on test data
y_predgbm = gbm.predict(X_test)

In [69]:
#MSE for GBM
mse_gbm = mean_squared_error(y_test, y_predgbm)
mse_gbm

0.14632775162943512

## Random Forest

In [70]:
#random split for train-test data
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2,random_state=1234)

In [71]:
from sklearn.ensemble import RandomForestRegressor

# Creating a Random Forest Regressor instance
rf_regressor = RandomForestRegressor(n_estimators=100, max_depth=20, max_features= "log2", random_state=42)

# Training the model
rf_regressor.fit(X_train, y_train)

# predicting on test
y_predrf = rf_regressor.predict(X_test)

# MSE for Random Forest
mserf = mean_squared_error(y_test, y_predrf)

In [72]:
X_test.head()

year  yearly_avg_snow  yearly_max_temp  yearly_min_temp  yearly_avg_temp  \
585  2019         7.970000             90.2              0.9        42.965833   
29   2015         0.708333             94.4             21.5        62.676667   
270  2016         0.614167             91.1             17.3        58.365000   
100  2014         0.216667             93.3             25.7        64.366667   
278  2012         0.000000             94.1             35.5        66.460000   

     yearly_avg_prcp  yearly_min_prcp  yearly_max_prcp  total pop  \
585         1.085000             0.31             3.95     571502   
29          5.410000             1.26            11.79    2929305   
270         3.440833             0.63             7.91    6000269   
100         3.849167             2.46             6.57    9918670   
278         4.881667             1.48             8.28    2942684   

     in-state pop  state_encoded  
585        541255             48  
29        2847711              3  
270       5837460             23  
100       9644575              8  
278       2869184             22

In [73]:
mserf

0.10695543368689359

In [74]:
#MAE for random forest
from sklearn.metrics import mean_absolute_error
mae = mean_absolute_error(y_test,y_predrf)
print(mae)

0.23247561613192894


## Models mse

Radom Forest: 0.1062 (random split)

ARIMA : 0.1318 (mean of each state)

GBM: 0.1493 (random split)

SVM: 0.1802 (random split)

NN: 0.3109 (random split) Data loss (4541636.17)

CNN: Data loss (10751.13)

Linear reg: 0.3775 (random split)

LSDM : 0.3775 (random split)

Desicion tree: 0.5027 (random split)


In [75]:
#END